ogre2/include/ignition/rendering/ogre2/Ogre2BoundingBoxCamera.hh

```diff
@@ -0,0 +1,190 @@
+      public: virtual void VisibleBoundingBoxes();
+
+      /// \brief Get the 3D bounding boxes after checking their visiblity
+      public: virtual void BoundingBoxes3D();
+
+      /// \brief Get minimal bounding box of the mesh by projecting the 3d
+      /// vertices of the vertex buffer to 2d then get the min & max of x & y
+      /// \param[in] mesh Mesh of the item to get its minimal bbox
+      /// \param[in] viewMatrix Camera view matrix
+      /// \param[in] projMatrix Camera projection matrix
+      /// \param[out] minVertex Minimum of projected x & y & z of the vertices
+      /// \param[out] maxVertex Maximum of projected x & y & z of the vertices
+      /// \param[in] position position of the item in the world space
+      /// \param[in] oreintation rotation of the item
+      /// \param[in] scale scale of the item
+      public: virtual void MeshMinimalBox(
+          const Ogre::MeshPtr mesh,
+          const Ogre::Matrix4 &viewMatrix,
+          const Ogre::Matrix4 &projMatrix,
+          Ogre::Vector3 &minVertex,
+          Ogre::Vector3 &maxVertex,
+          const Ogre::Vector3 &position,
+          const Ogre::Quaternion &oreintation,
+          const Ogre::Vector3 &scale
+          );
+
+      /// \brief Draw bounding box on the givin image
+      /// \param[in] data buffer contains the image data
+      /// \param[in] box bounding box to draw
+      public: virtual void DrawBoundingBox(unsigned char *_data,
+        const BoundingBox &_box) override;
+
+      /// \brief Convert from clip coord (after projection) to screen coord.
+      /// \param[inout] minVertex min vertex in clip coord to screen coord
+      /// \param[inout] maxVertex max vertex in clip coord to screen coord
+      public: virtual void ConvertToScreenCoord(Ogre::Vector3 &minVertex,
+          Ogre::Vector3 &maxVertex);
+
+      /// \brief Mark the visible boxes by checking the ogre ids map and mark
+      /// the ogre id which appears in the map.
+      public: virtual void MarkVisibleBoxes();
+
+      /// \brief Merge a vector of 2D boxes, used in multi-links model.
+      /// \param[in] _boxes Vector of 2D boxes
+      /// \return Merged bounding box
+      public: virtual BoundingBox MergeBoxes2D(
+        std::vector<BoundingBox *> _boxes);
+
+      /// \brief Get the 3d vertices(in world coord.) of the item's that belongs
+      /// to the same parent (only used in multi-links models)
+      /// \param[in] _ogreIds vector of ogre ids that belongs to the same model
+      /// \param[out] _vertices vector of 3d vertices of the item
+      public: virtual void GetMeshVertices(
+        std::vector<uint32_t> ogreIds, std::vector<math::Vector3d> &_vertices);
+
+      /// \brief Merge a links's 2d boxes of multi links models
+      public: virtual void MergeMultiLinksModels2D();
+
+      /// \brief Merge a links's 3d boxes of multi links models
+      public: virtual void MergeMultiLinksModels3D();
```

Comment:
can we make functions that are not used outside this class `private`? Same applies to the `OgreBoundingBoxCamera` class